### PR TITLE
[FLINK-18759][tests] Add readme.md for TPC-DS tools

### PR DESCRIPTION
**Pull request by leonardBang**
## What is the purpose of the change

* This pull request adding readme.md for TPC-DS tools to help user understand the test better.

## Brief change log

  - Add readme.md in `tpcds-tool`
  - Minor cleanup in `tpcds-tool/data_generator.sh `

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by leonardBang** [comment]
cc @godfreyhe could you help review this pr?

File: flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# TPC-DS tool
+
+This directory contains the necessary scripts/files to run a TPC-DS test.
+
+## data_generator.sh scripts
+
+Scripts that to download the TPC-DS data generator and generate test data, the supported OS including GNU/Linux OS and Mac OS.
```
Comment (godfreyhe) [inline flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:7]: data_generator.sh <generator_dir> <scale_factor> < data_dir> <retry_scripts_dir>

<generator_dir>, the directory to save TPC-DS data generator.
....

File: flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+* parameter 1: generator_dir, the directory to save TPC-DS data generator.
+* parameter 2: scale_factor, the TPC-DS test scale factor, we use the validation scale factor 1 (1G) by default.
```
Comment (godfreyhe) [inline flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:10]: Scale factor indicates raw data size in GB,  we use the validation scale factor 1 (1G) by default.

File: flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# TPC-DS tool
+
+This directory contains the necessary scripts/files to run a TPC-DS test.
```
Comment (godfreyhe) [inline flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:3]: add an official link: http://www.tpc.org/tpcds ?

File: flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+## data_generator.sh scripts
+
+Scripts that to download the TPC-DS data generator and generate test data, the supported OS including GNU/Linux OS and Mac OS.
+
+* parameter 1: generator_dir, the directory to save TPC-DS data generator.
+* parameter 2: scale_factor, the TPC-DS test scale factor, we use the validation scale factor 1 (1G) by default.
+* parameter 3: data_dir, the directory to save TPC-DS test data.
+* parameter 4: retry_scripts_dir, the directory which contains some common scripts for end-to end tests, the retry scripts will be used when download the TPC-DS data generator.
```
Comment (godfreyhe) [inline flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:12]: this is an optional argument ? btw, the name and the description do not match

File: flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+Directory that contains all TPC-DS queries (103 queries in total), the corresponding data scale factor is 1G, these queries comes from TPC-DS standard specification v2.11.0 [1].
+
+*Note:* The scale factor of queries should match with test data's scale factor.
+    
+## answer_set directory
+
+Directory that contains all answers for per query, the answer content of query may has different order according to the system's null order.
+
+* (1) Flink SQL keeps null first in ASC order, keeps null last in DESC order, we choose corresponding answer file when compare the answer.
+* (2) For query 8, 14a, 18, 70 and 77, the decimal precision of answer set is too low and unreasonable, we compare result with result from SQL server, they can strictly match.
```
Comment (godfreyhe) [inline flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:25]: improve this line ?

File: flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
 
-Scripts that to download the TPC-DS data generator and generate test data, the supported OS including GNU/Linux OS and Mac OS.
-
-* parameter 1: generator_dir, the directory to save TPC-DS data generator.
-* parameter 2: scale_factor, the TPC-DS test scale factor, we use the validation scale factor 1 (1G) by default.
-* parameter 3: data_dir, the directory to save TPC-DS test data.
-* parameter 4: retry_scripts_dir, the directory which contains some common scripts for end-to end tests, the retry scripts will be used when download the TPC-DS data generator.  
+Scripts to download the TPC-DS data generator and generate test data, the supported OS including GNU/Linux OS and Mac OS.
+```
+ data_generator.sh <generatorDir> <scaleFactor> <outputDataDir> <commonScriptsDir>
```
Comment (godfreyhe) [inline flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:9]: give some examples?

Comment (leonardBang) [reply flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:9]: `data_generator.sh` scripts is internally used by `test_tpcds.sh`, and it's only used by `test_tpcds.sh`.
It's still difficult for users to run a TPC-DS bench mark without `test_tpcds.sh` even I give an example.
So, I give an example of `test_tpcds.sh` and also post the use position in` test_tpcds.sh`, HDYT?

File: flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md [hunk 7]
```diff
@@ -11,13 +11,28 @@ Scripts to download the TPC-DS data generator and generate test data, the suppor
...
+$ cd /Users/demo/flink                                                                                   # go to the fFink project
+$ mvn compile -DskipTests=true                                                                          # compile the Flink source code
+$ export FLINK_DIR=/Users/demo/flink/flink-dist/target/flink-${flink-version}-bin/flink-${flink-version}      # set Flink distribution directory, the ${flink-version} is the compiled Flink version
+$ export END_TO_END_DIR=/Users/demo/flink/flink-end-to-end-tests                                          # set end to end tests directory
+$ mkdir -p /Users/demo/dir_to_save_genarator
+$ mkdir -p /Users/demo/dir_to_save_data
+$ cd /Users/demo/flink/flink-end-to-end-tests/tpcds-tool
+$ sh data_generator.sh /Users/demo/dir_to_save_genarator 1 /Users/demo/dir_to_save_data /Users/demo/flink/flink-end-to-end-tests/test-scripts
+```
+The downloaded generator will be saved to `/Users/demo/dir_to_save_genarator`, the generated data will be saved to `Users/demo/dir_to_save_data`.
```
Comment (godfreyhe) [inline flink-end-to-end-tests/flink-tpcds-test/tpcds-tool/README.md:27]: typo: `Users/demo`

**Review by godfreyhe — COMMENTED** [review]
Thanks for the update

**Review by godfreyhe — APPROVED** [review]
Thanks for the update, LGTM, +1 to merge

**Pull request merged**